linera-sdk/src/lib.rs

```diff
@@ -219,6 +219,19 @@ pub trait Contract: WithContractAbi + ContractAbi + Send + Sized {
         forwarded_sessions: Vec<SessionId>,
     ) -> Result<SessionCallOutcome<Self::Message, Self::Response, Self::SessionState>, Self::Error>;
 
+    /// Finishes the execution of the current transaction.
+    ///
+    /// This is called once before a transaction ends, to allow all applications that participated
+    /// in the transaction to perform any final operations, and optionally it may also cancel the
+    /// transaction if there are any pendencies.
```

Comment:
This should probably explain that if you override it, you need to call `store` yourself, if you want to persist the contract state.

Reply:
Good point. I tried to describe that, let me know what you think.